### PR TITLE
Disable sending multiple notifications for same event

### PR DESCRIPTION
**Pull request by Devorein**
### <samp>🤖 Generated by Copilot at bb45d4a</samp>

No summary available (Limit exceeded: required to process 126795 tokens, but only 50000 are allowed per call)

### CODE REVIEW

File: lib/notifications/proposals/createProposalNotifications.ts [hunk 1]
```diff
@@ -117,8 +117,9 @@ export async function createProposalNotifications(webhookData: {
         const accessibleProposalCategoryIds = accessibleProposalCategories.map(({ id }) => id);
 
         const isAuthor = proposalAuthorIds.includes(spaceRole.userId);
-        const isReviewer = proposalReviewerIds.includes(spaceRole.userId);
-        const isReviewerRole = proposalReviewerRoleIds.some((roleId) => roleIds.includes(roleId));
+        const isReviewer =
+          proposalReviewerIds.includes(spaceRole.userId) ||
+          proposalReviewerRoleIds.some((roleId) => roleIds.includes(roleId));
```
Comment (mattcasey) [inline lib/notifications/proposals/createProposalNotifications.ts:122]: the previous code was easier to read

File: lib/notifications/polls/createPollNotifications.ts [hunk 2]
```diff
@@ -82,18 +75,12 @@ export async function createPollNotifications(webhookData: {
...
-                  postCategories: [vote.post.category],
-                  userId: spaceUserId
-                });
+          const postPermission = await permissionClient.client.forum.computePostPermissions({
+            resourceId: vote.post.id,
+            userId: spaceUserId
+          });
 
-          if (categories.length !== 0 && categories[0].permissions.comment_posts && vote.author.id !== spaceUserId) {
+          if (postPermission.add_comment) {
```
Comment (mattcasey) [inline lib/notifications/polls/createPollNotifications.ts:83]: dont we still need to check the vote author is not the user?

File: lib/notifications/polls/createPollNotifications.ts [hunk 3]
```diff
@@ -56,19 +50,18 @@ export async function createPollNotifications(webhookData: {
...
-              : await premiumPermissionsApiClient.pages.computePagePermissions({
-                  resourceId: vote.page.id,
-                  userId: spaceUserId
-                });
-          if (pagePermission.comment && vote.author.id !== spaceUserId) {
+          const pagePermission = await permissionClient.client.pages.computePagePermissions({
+            resourceId: vote.page.id,
+            userId: spaceUserId
+          });
+          if (pagePermission.comment) {
```
Comment (mattcasey) [inline lib/notifications/polls/createPollNotifications.ts:64]: dont we need to check the vote author?

File: lib/notifications/polls/createPollNotifications.ts [hunk 4]
```diff
@@ -22,28 +21,23 @@ export async function createPollNotifications(webhookData: {
           id: voteId,
           status: 'InProgress'
         },
-        include: {
+        select: {
+          spaceId: true,
           page: {
-            select: { id: true, path: true, title: true }
+            select: { id: true }
```
Comment (mattcasey) [inline lib/notifications/polls/createPollNotifications.ts:27]: looks like you dont need to join on the tables, you can just select createdBy, pageId and postId like you did with space?

**Review by mattcasey — CHANGES_REQUESTED** [review]
looks pretty good but see questions

File: lib/notifications/proposals/createProposalNotifications.ts (hunk 1 above)
Comment (Devorein) [reply lib/notifications/proposals/createProposalNotifications.ts:122]: `isReviewer` should ideally check if the user is a reviewer or not, not just whether he is a reviewer based on direct user id reference neglecting the role.

File: lib/notifications/polls/createPollNotifications.ts (hunk 2 above)
Comment (Devorein) [reply lib/notifications/polls/createPollNotifications.ts:83]: That's done on line 51, we remove the poll author from the `spaceUserIds`

File: lib/notifications/polls/createPollNotifications.ts (hunk 3 above)
Comment (Devorein) [reply lib/notifications/polls/createPollNotifications.ts:64]: That's done on line 51, we remove the poll author from the `spaceUserIds`

File: lib/notifications/polls/createPollNotifications.ts (hunk 4 above)
Comment (Devorein) [reply lib/notifications/polls/createPollNotifications.ts:27]: Nice one. updated code.

**Pull request merged**